src/Workspaces/Remote/ServiceHub/Host/RemoteExperimentationService.cs

```diff
@@ -24,7 +24,15 @@ public bool IsExperimentEnabled(string experimentName)
         {
             // may return null in tests
             var assetSource = RemoteWorkspaceManager.Default.TryGetAssetSource();
-            return assetSource?.IsExperimentEnabledAsync(experimentName, CancellationToken.None).Result ?? false;
+            if (assetSource is null)
+                return false;
+
+            var isEnabledValueTask = assetSource.IsExperimentEnabledAsync(experimentName, CancellationToken.None);
+            if (isEnabledValueTask.IsCompleted)
+                return isEnabledValueTask.Result;
+
+            isEnabledValueTask.Preserve();
```

Comment:
what does this do?

Reply:
For ValueTask backed by constants or Task, nothing. But for ones backed by IValueTaskSource (which can pool), registers a continuation that calls GetResult to return the source lease.

Reply:
➡️ Changed the code to use `Forget` instead of `Preserve`, plus the commit links to microsoft/vs-threading#556.